Library/Homebrew/cleanup.rb

```diff
@@ -150,7 +150,7 @@ def self.install_formula_clean!(f)
 
     def periodic_clean_due?
       return false if ENV["HOMEBREW_NO_INSTALL_CLEANUP"]
-      return true unless PERIODIC_CLEAN_FILE.exist?
+      return HOMEBREW_CACHE.birthtime < CLEANUP_DEFAULT_DAYS.days.ago unless PERIODIC_CLEAN_FILE.exist?
```

Comment:
Could you turn this into an `if`/`else` instead? Thanks ❤️

Reply:
Just as soon as I figure out how to get around `birthtime` not being implemented in Linux…

Reply:
Oh no! `mtime` may be sufficient?

Reply:
Could even use that to replace `PERIODIC_CLEAN_FILE` entirely if it works and just `touch` the directory instead.

Reply:
Except that a directory's mtime and ctime are updated when files are added or removed directly within it. What we could do is (if `.cleaned` is missing) check `(HOMEBREW_REPOSITORY/".LICENSE.txt").mtime`, a file that's rarely changed and therefore a likely indicator of when Homebrew was installed.

Reply:
Could check `mtime` and if it's empty? If it's empty then just look at the modified time (or don't bother cleaning at all).

Reply:
like that?
```ruby
if PERIODIC_CLEAN_FILE.exist?
  PERIODIC_CLEAN_FILE.mtime < CLEANUP_DEFAULT_DAYS.days.ago
else
  file = (HOMEBREW_REPOSITORY/"Library/Taps/homebrew/homebrew-core/LICENSE.txt")
  if file.exist?
    file.mtime < CLEANUP_DEFAULT_DAYS.days.ago
  else
    ?
  end
end
```

however, what should we do if `LICENSE` not exist?

Reply:
The issue at hand is this: if the `.cleaned` file does not exist, where can we find a date that more or less reflects when Homebrew was first installed that a) is unlikely to change in the normal course of Homebrew updating itself, and b) will still be available during CI tests? (Turns out that `LICENSE.txt` isn't available when `test-bot` runs its tap tests, for which only  `{tmpdir}/prefix/Library/Taps/homebrew/homebrew-core/Formula/testball.rb` exists.) However, `HOMEBREW_REPOSITORY/"Library/Taps"` seems to fit the bill, since its mtime will only change if a third-party tap is added.

Reply:
@EricFromCanada sorry to yo-yo I think the previous `.cleaned` solution in https://github.com/Homebrew/install/pull/215 may be the best after all 😭.